### PR TITLE
Walletconnect improvements and fix

### PR DESCRIPTION
**Pull request by alexkeating**
### Description

I was unable to push directly to the existing branch  on #387. This pr should hook up i18n and add reload on wallet and chain change. I left a comment on #387 about the account center and I think " Cleanup unused imports, remove outdated/unused methods, fix any TS issues" is handled, but please let me know if I missed anything

### CODE REVIEW

**Comment by mds1** [comment]
About to review this, and just to clarify, all checkboxes in #387 should be done with the exception of the account center? And to confirm, I am ok with not using account center.

Pasting the checklist from #387 (along with what I think is their status in this PR) here for convenience:
- [x] Upgrade to v2 of onboard.js
- [x] Configure it to support popular wallets
- [x] Save most recently connected wallet to cache and try auto-connecting on page load
- [x] Configure i18n support: https://docs.blocknative.com/onboard/core (cc @garyghayrat, would love if you could push a commit that implements this)
- [x] Ensure we still refresh the page on wallet or network change
- [ ] Consider using the `accountCenter` feature to close https://github.com/ScopeLift/umbra-protocol/issues/315 — may do this in a future PR, not part of this PR
- [ x] Cleanup unused imports, remove outdated/unused methods, fix any TS issues

File: frontend/package.json [hunk 1]
```diff
@@ -17,13 +17,19 @@
     "precommit": "lint-staged"
   },
   "dependencies": {
+    "@ledgerhq/devices": "^7.0.0",
```
Comment (mds1) [inline frontend/package.json:20]: _(arbitrary comment location to start a thread)_

If I click on my address, it shows the below. We should probably show nothing / make that not clickable, and later make it clickable as part of https://github.com/ScopeLift/umbra-protocol/issues/315

image

File: frontend/package.json [hunk 2]
```diff
@@ -17,13 +17,19 @@
...
     "@metamask/jazzicon": "^2.0.0",
     "@quasar/extras": "^1.0.0",
     "@umbra/umbra-js": "^0.1.0",
     "@uniswap/token-lists": "^1.0.0-beta.19",
     "@vue/composition-api": "^0.6.4",
+    "@web3-onboard/coinbase": "^2.1.1",
```
Comment (mds1) [inline frontend/package.json:26]: _(arbitrary comment location to start a thread)_

Just noting that another wallet-related issue we have is #374, which we can do after this PR

File: frontend/src/store/wallet.ts [hunk 3]
```diff
@@ -1,26 +1,42 @@
...
 import useSettingsStore from 'src/store/settings';
-
+import enLocal from 'src/i18n/locales/en-us.json';
+import chLocal from 'src/i18n/locales/zh-cn.json';
+
+// Wallet configurations.
+const injected = injectedModule();
+const walletConnect = walletConnectModule({
+  qrcodeModalOptions: { mobileLinks: ['rainbow', 'metamask', 'argent', 'trust', 'imtoken', 'pillar'] },
+});
```
[the remark this answers is not in the document]
Comment (alexkeating) [reply frontend/src/store/wallet.ts:36]: I will remove

File: frontend/package.json (hunk 1 above)
Comment (alexkeating) [reply frontend/package.json:20]: That makes sense will do

**Review by apbendi — COMMENTED** [review]
Overall working really well in QA testing @alexkeating, nice work! One thing I noticed is that the network switcher does not work with walletconnect, at least not with Rainbow Wallet on mobile which I was using to test.

File: frontend/src/store/wallet.ts [hunk 4]
```diff
@@ -1,26 +1,40 @@
...
 import { UmbraApi } from 'src/utils/umbra-api';
 import { getChainById } from 'src/utils/utils';
 import useSettingsStore from 'src/store/settings';
-
+import enLocal from 'src/i18n/locales/en-us.json';
+import chLocal from 'src/i18n/locales/zh-cn.json';
+
+// Wallet configurations.
+const injected = injectedModule();
+const walletConnect = walletConnectModule();
```
Comment (alexkeating) [inline frontend/src/store/wallet.ts:34]: Notes on walletconnect. It looks like onboard wallet connect does not support wallet_switchEthereumChain. Looking into potential alternatives.

https://github.com/blocknative/web3-onboard/blob/v2-web3-onboard-develop/packages/walletconnect/src/index.ts#L183

File: frontend/package.json (hunk 1 above)
Comment (alexkeating) [reply frontend/package.json:20]: Resolved

File: frontend/src/store/wallet.ts (hunk 4 above)
Comment (alexkeating) [reply frontend/src/store/wallet.ts:34]: Made an [issue](https://github.com/blocknative/web3-onboard/issues/1313) and hope to get this implemented on there side.

Comment (mds1) [reply frontend/src/store/wallet.ts:34]: Awesome thank you

File: frontend/package.json (hunk 1 above)
Comment (mds1) [reply frontend/package.json:20]: Nit, still looks like hovering over it shows the "cursor" style pointer and not the default one. Other than that this PR LGTM!

Comment (alexkeating) [reply frontend/package.json:20]: Good catch! I pushed up a fix

**Review by mds1 — APPROVED** [review]
Ready to merge IMO, pending any QA stuff from @apbendi. CI failure is unrelated and is fixed in #401

**Review by apbendi — APPROVED** [review]
Looking and working great, with the exception of the previously reported network switching issue which is tracked in #404. Good to merge!

**Pull request merged**